### ISSUE TITLE
Upserting with client generated ID updates ID

### ISSUE THREAD

**Issue by ivanvanderbyl** [ent/ent#1826]
When using custom IDs, such as `uuid.New` or `ksuid.New` as the value, and not using a database generated ID, Upserts change the ID (primary key) of the row on conflict when attempting to return the ID, even when using `Exec()` instead of `ID()`.

This seems to be due to the clause `id = excluded.id`, which is used to return an ID. This could potentially be changed to `id = table_name.id`

- [x] The issue is present in the latest release.
- [x] I have searched the [issues](https://github.com/ent/ent/issues) of this repository and believe that this is not a duplicate.

So far I've been able to workaround this with:

```go
sql.ResolveWith(func(s *sql.UpdateSet) {
	s.Set("id", sql.Expr(s.Table().C("id")))
}),
```

**Comment by a8m** [comment]
Thanks for reporting this @ivanvanderbyl 😃 

It happens only when you use `UpdateNewValues`, right? If so, I already have a patch for that.

**Comment by a8m** [comment]
Hey @ivanvanderbyl, I just landed https://github.com/ent/ent/pull/1836 that should fix this issue.

Can you please confirm that the latest master version solved your problem and update me? I plan to release a new patch version in the upcoming days. Thanks 🙏 

```go
go get entgo.io/ent@master
```

**Comment by ivanvanderbyl** [comment]
> Hey @ivanvanderbyl, I just landed #1836 that should fix this issue.
> 
> Can you please confirm that the latest master version solved your problem and update me? I plan to release a new patch version in the upcoming days. Thanks 🙏
> 
> ```go
> go get entgo.io/ent@master
> ```

Missed this, thanks for fixing, I'll try it out.
